src/partition/repart.c

```diff
@@ -4985,6 +5200,25 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
                 }
 
+                case ARG_CERTIFICATE_FILE: {
+                        _cleanup_(erase_and_freep) char *cert = NULL;
+                        size_t n = 0;
+
+                        r = read_full_file_full(
+                                        AT_FDCWD, optarg, UINT64_MAX, SIZE_MAX,
+                                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                                        NULL,
+                                        &cert, &n);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to read certificate file '%s': %m", optarg);
+
+                        erase_and_free(arg_certificate);
+                        r = parse_x509_certificate(cert, n, &arg_certificate, &arg_certificate_size);
+                        if (r < 0)
+                                return r;
+                        break;
+                }
```

Comment:
hmm, so we need a separate parser for the private key. You cannot really overload --key-file= for this, since encryption requires a symmetric key (i.e. random binary data) while this signing stuff requires an asymmetric PEM key (i.e. strictly structured data). and sometimes you want to build an image with an encrypted /home and a signed root, but you basically disallow that by overloading it.

Hence, I'd call the certificate switch "--certificate=" and the private key switch "--private-key=". And then leave --key-file= only for the symetric LUKS stuff.